rasa_core/trackers.py

```diff
@@ -212,14 +220,29 @@ def generate_all_prior_trackers(self):
         The resulting array is representing
         the trackers before each action."""
 
-        tracker = self.init_copy()
+        if_featurized = []
+        idx_of_failed = None
+        for i, event in enumerate(self.applied_events()):
+            if isinstance(event, ActionExecutionFailed):
+                if_featurized.append(True)
+                idx_of_failed = i
+            elif isinstance(event, FormIsBack):
+                if idx_of_failed is not None:
+                    if_featurized[idx_of_failed:] = [False] * len(if_featurized[idx_of_failed:])
+                if_featurized.append(False)
+            else:
+                if_featurized.append(if_featurized[-1] if if_featurized else True)
 
-        for event in self.applied_events():
+        tracker = self.init_copy()
+        for i, event in enumerate(self.applied_events()):
             if isinstance(event, ActionExecuted):
-                yield tracker
+                if tracker.active_form is None or (tracker.active_form == self.active_form and if_featurized[i]):
```

Comment:
what if the user does one form first and then another one, `self.active_form == self.active_form` will be falls for any `chitchat` in the first form, so we ignore that during training?